### PR TITLE
Create Api forget password and reset it

### PR DESCRIPTION
**Pull request by AhmedQeshta**
## Create Api forget the password and reset it

👍🏻  API one :
`/api/v1/gyms/reset/password`
> body request is
```json
{
     "password": ".....",
     "confirmPassword": ".....",
     "token": "....."
}
``` 

👍🏻  API two :
`/api/v1/gyms/forget/password`
> body request is
```json
{
     "email": "....."
}
```

### CODE REVIEW

File: server/utils/validation/resetPasswordSchema.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-useless-escape */
+const Joi = require('joi');
```
Comment (LinaYahya) [inline server/utils/validation/resetPasswordSchema.ts:2]: import instead of require

File: server/utils/validation/resetPasswordSchema.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-useless-escape */
```
Comment (LinaYahya) [inline server/utils/validation/resetPasswordSchema.ts:1]:
```suggestion
```

File: server/utils/validation/forgetPasswordSchema.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-useless-escape */
+const Joi = require('joi');
```
Comment (LinaYahya) [inline server/utils/validation/forgetPasswordSchema.ts:2]:
```suggestion
const Joi = require('joi'); // import instead
```

File: server/controllers/auth/resetPassword.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      .cookie('token', newToken, {
+        maxAge: 900000,
+      })
+      .json({
+        message: 'تم تحديث كلمة المرور بنجاح',
+        payload,
+      });
+  } catch (error: any) {
+    if (error.name === 'ValidationError') {
+      next(new CustomError(error.message, 400));
```
Comment (LinaYahya) [inline server/controllers/auth/resetPassword.ts:48]:
```suggestion
      return next(new CustomError(error.message, 400));
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by AFahgan — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**